txpool/txpool.go

```diff
@@ -117,6 +117,8 @@ type TxPool struct {
 
 	// Indicates which txpool operator commands should be implemented
 	proto.UnimplementedTxnPoolOperatorServer
+
+	metrics *Metrics
```

Comment:
Metrics is a specific type of object because it is needed in many of the modules of the code. Then, it is usually set as a global variable like [this](https://github.com/ethereum/go-ethereum/blob/d8ff53dfb8a516f47db37dbc7fd7ad18a1e8a125/eth/downloader/metrics.go#L26). If you use metrics this way it will simplify a lot the PR since you will not need to move around a Metrics type.

Reply:
@ferranbt I felt like having a separate metric type for each subsystem (**_consensusMetrics_**,**_txpoolMetrics_**,**_p2pMetrics_**) would be more cleaner and give us more control down the line. But I do agree with you that it makes the PR more simpler.
@lazartravica @Vuksan Whats your take ?

Reply:
@munna0908 I'd be happy to discuss more, I am leaning more towards what Ferran said.

If you could argument what kind of control would be given by current implementation, that would be excellent.

My main point is that here we should try to make non-functional features such as Metrics make as less changes to the signatures of our methods as possible. In the end this would simplify testing as well, not only usage.

Reply:
@lazartravica As discussed, i feel this approach will make future enhancements very easily(when we would want to support different instrumentations)and it will not be lot of overhead for writing other unit test cases, since the constructor to individual metric type can return a nilmetric if the instrumentation type is set to nil.
Looking forward for your inputs @ferranbt

Reply:
@munna0908 Can you please provide some examples about this? Got it for the tests.

Reply:
@lazartravica  For example, for consensus metric type, currently we have a [function](https://github.com/0xPolygon/polygon-sdk/blob/34c58d0340c500938a6487846884eb2e8e9d7136/consensus/metrics.go#L21) to return prometheus metrics, likewise we can define functions to return other metrics like openMetrics etc.

Reply:
@munna0908 why wouldn't you be able to do it in the example that ferran linked in the first reply, I still don't understand?

Reply:
> @munna0908 why wouldn't you be able to do it in the example that ferran linked in the first reply, I still don't understand?

@lazartravica we can absolutely do it,but i feel it will not be very clean and it is cumbersome to maintain.